### PR TITLE
Enhance the bottom bar adaptability to navigation actions accross the app

### PR DESCRIPTION
**Pull request by mohagpt**
This PR introduces a feature enhancement to the QuickFix app's bottom navigation bar, allowing it to adapt based on the current navigation actions. The bottom bar will now dynamically update its state and items according to the top-level destinations in the app's navigation flow. Additionally, a comprehensive test has been added to ensure the new behavior works as expected.

Changes Made
- Feature Implementation:
Updated the bottom bar logic to adapt to navigation actions, ensuring the displayed items are consistent with the current route.
Refactored MainActivity, BottomNavigationMenu, and NavigationActions to support dynamic updates of the bottom bar.

- Testing:
Added a test case for the "Cancel" button on the Search screen:
Verified that clicking the "Cancel" button correctly navigates to the home screen.
Confirmed that the bottom bar updates accordingly when navigating to TopLevelDestinations.HOME.
The new test case ensures proper functionality of the adaptive bottom bar feature.

### CODE REVIEW

File: app/src/androidTest/java/com/arygm/quickfix/ui/search/QuickFixFinderScreenTest.kt
Comment (AnotherMedo) [inline app/src/androidTest/java/com/arygm/quickfix/ui/search/QuickFixFinderScreenTest.kt:1]: Drop these changes as they are unrelated to this pull request

**Review by AnotherMedo — CHANGES_REQUESTED** [review]
The bottom navigation bar still has some bugs that need to be addressed:

- The bottom navigation bar is still visible from the account configuration screen which should not be the case
- The icons have some text appearing on them that should not

File: app/src/androidTest/java/com/arygm/quickfix/ui/search/QuickFixFinderScreenTest.kt
Comment (mohagpt) [reply app/src/androidTest/java/com/arygm/quickfix/ui/search/QuickFixFinderScreenTest.kt:1]: For the first bug, i will correct it as it is fast to do. 
For the second one, it is actually not a bug as we decide to add these names only to the debug version for testing reasons, as this part of the code suggest : 
if (BuildConfig
                .DEBUG) { // when we'll want to generate the APK since it will be in release mode
              // normally this will not be executed
              if (model != null) {
                model.count = tab.route
              } // Set the count of the model
              countBackgroundColor = 0
            }

Comment (AnotherMedo) [reply app/src/androidTest/java/com/arygm/quickfix/ui/search/QuickFixFinderScreenTest.kt:1]: Sounds good! thanks for being so reactive :)

**Review by AnotherMedo — APPROVED** [review]
It all looks good to me now. Great job ! We are making great progress towards M2 :D

**Review by AnotherMedo — APPROVED** [review]
